CRM/Contact/Form/Contact.php

```diff
@@ -1192,47 +1192,22 @@ public function postProcess() {
    * @return bool
    *   true if data exists, false otherwise
    */
-  public static function blockDataExists(&$fields) {
+  public static function blockDataExists($fields): bool {
     if (!is_array($fields)) {
+      CRM_Core_Error::deprecatedWarning('support for invalid values will be dropped');
       return FALSE;
     }
 
-    static $skipFields = [
+    $dataFields = array_filter(array_diff_key($fields, array_fill_keys([
       'location_type_id',
       'is_primary',
       'phone_type_id',
       'provider_id',
       'country_id',
       'website_type_id',
       'master_id',
-    ];
-    foreach ($fields as $name => $value) {
-      $skipField = FALSE;
-      foreach ($skipFields as $skip) {
-        if (strpos("[$skip]", $name) !== FALSE) {
-          if ($name == 'phone') {
```

Comment:
@eileenmcnaughton this seems like a special case being removed. Was it irrelevant?

Reply:
I don't think so

Reply:
So it shouldn't have been removed?

Reply:
@colemanw sorry - I don't think it was relevant - it seemed like crufy based on them having phone_type_id in the skip fields array & then 'phone' would be in a str_contains on phone_type_id

Reply:
Ok so we don't have to worry about other fields named e.g. `phone` and `phone_numeric`?
Seems like this code might have been intended to catch both of those.

Reply:
@colemanw I think it pre-ceded that field exisitng - I mean we are talking about them casting array keys to strings before doing a string search. We don't expose phone_numeric in the UI in these screens so it's neither here nor there but I guess it if WAS submitted it should be saved - well maybe - but it isn't

Reply:
Ok